python/lsst/cp/pipe/cpTask.py

```diff
@@ -0,0 +1,594 @@
+
+        Parameters
+        ----------
+        pSet : daf.base.PropertySet
+           PropertySet to be translated to an integer-keyed dictionary
+        """
+        return {int(amp): gain for amp, gain in pSet.toDict().items()}
+
+    def testMethod(self):  # TODO: XXX remove this whole method, just useful for debugging/dev work
+        """TODO: Remove this docstring that only exists to remove flake8 error."""
+        self.log.warn("Test warning message")
+        print(self.config.requireAllEOTests)
+
+    # @pipeBase.timeMethod #xxx re-include this
+    def runEotestDirect(self, butler, run=None):
+        """
+        Generate calibration products using eotest algorithms.
+
+        Generate all calibration products possible using the vanilla eotest implementation,
+        given a butler for a TS8 (raft-test) repo. It can contain multiple runs, but must correspond to
+        only a single raft/RTM.
+
+        - Run all eotest tasks possible, using the butler to gather the data
+        - Write outputs in eotest format
+
+        In order to replicate the canonical eotest analysis, the tasks should be run in a specific order.
+        This is given/defined in the "Steps" section here:
+        http://lsst-camera.slac.stanford.edu/eTraveler/exp/LSST-CAMERA/
+               displayProcess.jsp?processPath=1179
+
+        But is replicated here for conveniece:
+        * 55Fe Analysis
+        * CCD Read Noise Analysis
+        * Bright Defects Analysis
+        * Dark Defects Analysis
+        * Traps Finding
+        * Dark Current                  X - will not be implemented here
+        * Charge Transfer Efficiencies
+        * Photo-response analysis       X - will not be implemented here
+        * Flat Pairs Analysis
+        * Photon Transfer Curve
+        * Quantum Efficiency            X - will not be implemented here
+
+        List of tasks that exist in the eotest package but aren't mentioned on the above link:
+        --------
+        # linearityTask()
+        # fe55CteTask()
+        # eperTask()
+        # crosstalkTask()
+        # persistenceTask()
+
+        # TODO: For each eotest task, find out what the standard raft testing does for the optional params.
+        i.e. many have optional params for gains, bias-frames etc - if we want bitwise identicallity then we
+        need to know what is typically provided to these tasks when the camera team runs this code.
+        This can probably be worked out from https://github.com/lsst-camera-dh/lcatr-harness
+        but it sounds like Jim Chiang doesn't recommend trying to do that.
+
+        Parameters
+        ----------
+        butler : daf.persistence.butler
+            Butler for the repo containg the eotest data to be used
+        run : string or int
+            Optional run number, to be used for repos containing multiple runs
+        """
+        self.log.info("Running eotest routines direct")
+        runs = butler.queryMetadata('raw', ['run'])
```

Comment:
I generally consider it a bad idea for variable names to differ by a single character, as it's easy to confuse them.

Reply:
Indeed, but this is why I `del` it as soon as I'm done with it. Given that, I think it's OK. Shout if you disagree.